packages/cozy-pouch-link/src/startReplication.js

```diff
@@ -208,3 +235,99 @@ export const replicateAllDocs = async ({ db, baseUrl, doctype, storage }) => {
+ * This function fetches documents in batches from a twake drive shared drive
+ * and replicates them to the local database, maintaining replication state
+ * to allow for incremental updates.
+ * We do not have an _all_docs view for shared drives, so we use the fetchChanges method.
+ *
+ * @param {Object} params - The parameters object
+ * @param {string} params.driveId - The unique identifier of the shared drive to replicate from
+ * @param {Object} params.pouch - The local PouchDB instance to replicate documents to
+ * @param {Object} params.storage - Storage interface for persisting replication state
+ * @param {string} params.doctype - The document type being replicated (e.g., 'io.cozy.files')
+ * @param {boolean} params.initialReplication - Whether this is an initial replication
+ * @param {CozyClient} params.client - CozyClient instance for fetching remote documents
+ * @returns {Promise<Array>} A promise that resolves to an array of all replicated documents
+ * @throws {Error} Throws an error if driveId is not provided
+ */
+export const sharedDriveReplicateAllDocs = async ({
+  driveId,
+  pouch,
+  storage,
+  initialReplication = false,
+  doctype,
+  client
+}) => {
+  if (!driveId) {
+    throw new Error('sharedDriveReplicateAllDocs: driveId is required')
+  }
+  let docs = []
+  let hasMore = true
+
+  let doctypeLastSequence = await storage.getDoctypeLastSequence(doctype)
+  while (hasMore) {
+    const { newLastSeq, results, pending } = await client
+      .collection('io.cozy.files', { driveId })
+      .fetchChanges(
+        {
+          include_docs: true,
+          ...(doctypeLastSequence ? { since: doctypeLastSequence } : {})
+        },
+        {
+          includeFilePath: false,
+          ...(initialReplication
+            ? { skipDeleted: true, skipTrashed: true }
+            : { skipDeleted: false, skipTrashed: false }),
+          limit: BATCH_SIZE
+        }
+      )
+
+    const toDelete = []
+    const toInsert = []
+    const allDocsWithDriveId = []
+    for (const doc of results) {
+      const docWithDriveId = doc.doc
+      docWithDriveId.driveId = driveId
+      allDocsWithDriveId.push(docWithDriveId)
+      if (doc.doc._deleted) {
+        toDelete.push(doc.doc)
+      } else {
+        toInsert.push(docWithDriveId)
+      }
+    }
+    // FIXME this is a workaround to allow to delete documents from the shared drive
+    // changes
+    // PouchDB.bulkDocs ignores _deleted documents with revision newer than the existing
+    // document
+    // The workaround is to get documents from PouchDB with correct revision and delete them
+    if (toDelete.length > 0) {
+      // lets try to find a document with the same _id in allDocsWithDriveId
+      for (const toDeleteDoc of toDelete) {
+        try {
+          const originalDoc = await pouch.get(toDeleteDoc._id)
+          if (originalDoc) {
+            await pouch.remove(originalDoc)
```

Comment:
since this works it is not blocking, but I think we should eventually do a bulkGet and then a bulkDelete, that would be much more efficient